### PR TITLE
Add cache groups

### PR DESCRIPTION
**Pull request by skeggse**
This lets developers partition their caches in case they're doing funky things with module resolution and want to avoid poisoning their cache with the wrong copy of the module.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,6 +43,10 @@ class MultiBuild {
    *  @param {Gulp} gulp - The gulp instance with which to register build tasks.
    *  @param {Array<String>} targets - The names of the targets to build, arbitrary identifiers.
    *    Must not contain spaces since they will be used to form the names of the build tasks.
+   *  @param {Map<String,Set<String>|String[]>|Object<String,Set<String>|String[]>} cacheGroups The
```
Comment (wearhere) [inline index.js:46]: Could you add an example to the README?

File: index.js [hunk 2]
```diff
@@ -42,29 +65,85 @@ class MultiBuild {
...
    * Builds all target bundles and registers dependencies on the files that comprise each bundle.
    *
    * @param {Function} done - Callback.
    */
   runAll(done) {
-    // We run the target tasks sequentially, so that each run can benefit from the cached AST from
-    // the previous runs.
-    const targetTasks = this._targets.map(MultiBuild.task);
-    runSequence.use(this._gulp).apply(undefined, targetTasks.concat(done));
+    // We run the groups in parallel, but each target tasks with a group sequentially, so that each
```
Comment (wearhere) [inline index.js:144]: nit: within a group, I think

**Review by wearhere — COMMENTED** [review]
neat!

File: index.js (hunk 2 above)
Comment (skeggse) [reply index.js:144]: I think either work, but I like within better

**Pull request merged**